core/go/internal/sequencer/coordinator/select_transaction.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright © 2025 Kaleido, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package coordinator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/LFDT-Paladin/paladin/common/go/pkg/i18n"
+	"github.com/LFDT-Paladin/paladin/common/go/pkg/log"
+	"github.com/LFDT-Paladin/paladin/core/internal/msgs"
+	"github.com/LFDT-Paladin/paladin/core/internal/sequencer/coordinator/transaction"
+	"github.com/google/uuid"
+)
+
+func (c *coordinator) selectNextTransactionToAssemble(ctx context.Context, event *TransactionStateTransitionEvent) error {
+	log.L(ctx).Trace("selecting next transaction to assemble")
+	txn, err := c.transactionSelector.SelectNextTransactionToAssemble(ctx, event)
+	if err != nil {
+		log.L(ctx).Errorf("error selecting transaction: %v", err)
+		return err
+	}
+	if txn == nil {
+		log.L(ctx).Info("no transaction found to process")
+		return nil
+	}
+
+	transactionSelectedEvent := &transaction.SelectedEvent{}
+	transactionSelectedEvent.TransactionID = txn.ID
+	err = txn.HandleEvent(ctx, transactionSelectedEvent)
+	return err
+
+}
+
+func (c *coordinator) AddTransactionToBackOfPool(ctx context.Context, txn *transaction.Transaction) {
+	c.pooledTransactionsMutex.Lock()
+	defer c.pooledTransactionsMutex.Unlock()
+	c.pooledTransactions = append(c.pooledTransactions, txn)
+}
+
+/* Functions of the TransactionPool interface required by the transactionSelector */
+func (c *coordinator) GetNextPooledTransaction(ctx context.Context) *transaction.Transaction {
```

Comment:
```suggestion
func (c *coordinator) PopNextPooledTransaction(ctx context.Context) *transaction.Transaction {
```

Reply:
Or could be remove... but think it needs to be clear this isn't a getter, but it mutates the queue by removing the first element.